client/src/components/MapGeocoderWrapper.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div v-show='visible' class='geocoder-content-expanded'>
+    <div class='geocoder' id='geocoder'></div>
+    <div class='search-button' @click='this.visible = !this.visible'>
+      <img src='@/assets/icons/search.svg' />
+    </div>
+  </div>
+  <div v-show='!visible' class='geocoder-content-collapsed'>
+    <div class='search-button' @click='this.visible = !this.visible'>
+      <img src='@/assets/icons/search.svg' />
+    </div>
+  </div>
+</template>
+
+<script lang='ts'>
+import { defineComponent, inject } from 'vue';
+import { State } from '../model/state';
+import { stateKey } from '../injection_keys';
+import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
+import '@mapbox/mapbox-gl-geocoder/dist/mapbox-gl-geocoder.css';
+import mapboxgl from 'mapbox-gl';
+
+/**
+ * Wrapper for Mapbox Geocoder.
+ *
+ * See details at https://github.com/mapbox/mapbox-gl-geocoder.
+ */
+export default defineComponent({
+  name: 'MapGeocoderWrapper',
+  setup() {
+    const state: State = inject(stateKey, State.default());
+
+    return {
+      state,
+    };
+  },
+  data() {
+    return {
+      geocoder: null as unknown as MapboxGeocoder,
```

Comment:
Same as above regarding `as unknown` here.

Reply:
This one actually is necessary if we want to have this field be of type MapboxGeocoder without assigning an actual MapboxGeocoder value on init. This is because the inferred type will just be null if we don't do typecasting here (there is no way to explicitly declare type for data fields AFAIK). 

Alternatively we could initialize just an empty geocoder but its sort of misleading because it will not be linked to the map at all or show up on the page since we create the element when the actual geocoder is initialized - let me know if you prefer this approach.

Reply:
What is the cost of having an empty geocoder? I think that sounds fine, as long as it's not some expensive operation.

Reply:
No cost - done!